### PR TITLE
Fix test.utils.assertRaisesRegexp() when Regex is a `str`

### PR DESCRIPTION
**Pull request by JordanP**

### CODE REVIEW

File: test/unit/utils.py [hunk 1]
```diff
@@ -46,9 +46,15 @@ def assertRaisesRegexp(expected_exception, expected_regexp,
         callable_obj(*args, **kwargs)
     except expected_exception as exc_value:
         if not re.search(expected_regexp, str(exc_value)):
+            # We accept both `string` and compiled regex object as 2nd
+            # argument to assertRaisesRegexp
+            if hasattr(expected_regexp, 'pattern'):
```
Comment (NicolasT) [inline test/unit/utils.py]: `pattern = getattr(expected_regexp, 'pattern', expected_regexp)`

Comment (JordanP) [inline test/unit/utils.py]: Yep.

**Comment by NicolasT** [comment]
LGTM (although I'm a bit surprised `re.search` accepts both `str`s as well as compiled regexes)

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/16/<h2>Build result: FAILURE</span></h2>[...truncated 6 lines...]Fetching upstream changes from https://github.com/scality/scality-sproxyd-client > git --version # timeout=10 > git -c core.askpass=true fetch --tags --progress https://github.com/scality/scality-sproxyd-client +refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse refs/remotes/origin/pr/9/merge^{commit} # timeout=10 > git rev-parse refs/remotes/origin/origin/pr/9/merge^{commit} # timeout=10Checking out Revision ee0ed7d444525603d50850ea9d2110336ff34cc8 (refs/remotes/origin/pr/9/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f ee0ed7d444525603d50850ea9d2110336ff34cc8First time build. Skipping changelog.Triggering scality-sproxyd-client-tox » py26,centos6-largeTriggering scality-sproxyd-client-tox » py27,ubuntu14-largeTriggering scality-sproxyd-client-tox » pep8,ubuntu14-largeTriggering scality-sproxyd-client-tox » py27,centos7-largeTriggering scality-sproxyd-client-tox » py27,ubuntu12-largeConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: All nodes of label ‘centos6-large’ are offlineConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: centos6-large-8cc is offlineConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: Waiting for next available executor on centos6-large-8ccscality-sproxyd-client-tox » py26,centos6-large completed with result SUCCESSscality-sproxyd-client-tox » py27,ubuntu14-large completed with result FAILUREscality-sproxyd-client-tox » pep8,ubuntu14-large completed with result FAILUREscality-sproxyd-client-tox » py27,centos7-large completed with result SUCCESSConfiguration scality-sproxyd-client-tox » py27,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-largescality-sproxyd-client-tox » py27,ubuntu12-large completed with result FAILURESetting status of a3ca9e1b423bf483ae8aab45d296a75764bfd243 to FAILURE with url https://37.187.159.67:5443/job/scality-sproxyd-client-tox/16/ and message: Merged build finished.
Test FAILed.

**Comment by JordanP** [comment]
```
In [4]: re.search("myregex", "salut myregex salut").group()
Out[4]: 'myregex'

In [5]: re.search(re.compile("myregex"), "salut myregex salut").group()
Out[5]: 'myregex'
```

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/17/<h2>Build result: FAILURE</span></h2>[...truncated 6 lines...]Fetching upstream changes from https://github.com/scality/scality-sproxyd-client > git --version # timeout=10 > git -c core.askpass=true fetch --tags --progress https://github.com/scality/scality-sproxyd-client +refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse refs/remotes/origin/pr/9/merge^{commit} # timeout=10 > git rev-parse refs/remotes/origin/origin/pr/9/merge^{commit} # timeout=10Checking out Revision 3994334f5935beffc4e43f679a5d6ee3858677ff (refs/remotes/origin/pr/9/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f 3994334f5935beffc4e43f679a5d6ee3858677ff > git rev-list ee0ed7d444525603d50850ea9d2110336ff34cc8 # timeout=10First time build. Skipping changelog.Triggering scality-sproxyd-client-tox » py26,centos6-largeTriggering scality-sproxyd-client-tox » py27,ubuntu14-largeTriggering scality-sproxyd-client-tox » pep8,ubuntu14-largeTriggering scality-sproxyd-client-tox » py27,centos7-largeTriggering scality-sproxyd-client-tox » py27,ubuntu12-largeConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: centos6-large-8cc is offlineConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: All nodes of label ‘centos6-large’ are offlineConfiguration scality-sproxyd-client-tox » py26,centos6-large is still in the queue: Waiting for next available executor on centos6-largescality-sproxyd-client-tox » py26,centos6-large completed with result FAILUREscality-sproxyd-client-tox » py27,ubuntu14-large completed with result FAILUREscality-sproxyd-client-tox » pep8,ubuntu14-large completed with result FAILUREscality-sproxyd-client-tox » py27,centos7-large completed with result FAILUREscality-sproxyd-client-tox » py27,ubuntu12-large completed with result FAILURESetting status of a2888f49d3e341231c17045be9170848b8bbf880 to FAILURE with url https://37.187.159.67:5443/job/scality-sproxyd-client-tox/17/ and message: Merged build finished.
Test FAILed.

**Pull request merged**